### PR TITLE
Made  loadMore smoother

### PR DESCRIPTION
**Pull request by ZeRoGerc**

### CODE REVIEW

File: app/src/main/java/ly/loud/loudly/application/models/LoadMoreStrategyModel.java [hunk 1]
```diff
@@ -47,9 +47,16 @@ public TimeInterval getCurrentTimeInterval() {
...
+        if (!isTimeIntervalIsInfinity()) {
             loadLastDays *= 2;
             updateTimeInterval();
         }
     }
+
+    /**
+     * Return true if it's likely that no other posts could been loaded if timeInterval increased
+     */
+    public boolean isTimeIntervalIsInfinity() {
```
Comment (danilkolikov) [inline app/src/main/java/ly/loud/loudly/application/models/LoadMoreStrategyModel.java]: isTimeIntervalIsInfinity? Too many "is", I think

File: app/src/main/java/ly/loud/loudly/ui/feed/FeedFragment.java [hunk 2]
```diff
@@ -182,18 +196,45 @@ public void onDeleteClick(@NonNull PlainPost post) {
...
+    private void setLoadMoreListener() {
+        final StaggeredGridLayoutManager manager = (StaggeredGridLayoutManager) feedRecyclerView.getLayoutManager();
+        int[] cache = new int[] {RecyclerView.NO_POSITION, RecyclerView.NO_POSITION};
+
+        feedRecyclerView.addOnScrollListener(new RecyclerView.OnScrollListener() {
+            @Override
+            public void onScrolled(@NonNull RecyclerView recyclerView, int dx, int dy) {
+                super.onScrolled(recyclerView, dx, dy);
+                manager.findLastVisibleItemPositions(cache);
+                for (int i = 0; i < 2; i++) {
```
Comment (danilkolikov) [inline app/src/main/java/ly/loud/loudly/ui/feed/FeedFragment.java]: I can' understand this cycle, add comment pls

Comment (ZeRoGerc) [inline app/src/main/java/ly/loud/loudly/ui/feed/FeedFragment.java]: 2 - is a max span number

File: app/src/main/java/ly/loud/loudly/ui/feed/FeedFragment.java [hunk 3]
```diff
@@ -182,18 +201,47 @@ public void onDeleteClick(@NonNull PlainPost post) {
...
+            public void onScrolled(@NonNull RecyclerView recyclerView, int dx, int dy) {
+                super.onScrolled(recyclerView, dx, dy);
+                manager.findLastVisibleItemPositions(cache);
+                for (int i = 0; i < MAX_SPAN_NUMBER; i++) {
+                    // cache[i] - is number of post in i-th span
+                    if (cache[i] == adapter.getItemCount() - 1) {
+                        needMoreItems();
+                    }
+                }
+
```
Comment (danilkolikov) [inline app/src/main/java/ly/loud/loudly/ui/feed/FeedFragment.java]: empty line

**Comment by danilkolikov** [comment]
ok

**Pull request merged**